pkg/cmd/template/cmd_assert_validate_test.go

```diff
@@ -0,0 +1,273 @@
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(dataValuesYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("has incorrect number of args in a validation", func(t *testing.T) {
+			opts := cmdtpl.NewOptions()
+			dataValuesYAML := `#@data/values
+---
+#@assert/validate (lambda v: v,)
+foo: bar
+`
+
+			expectedErr := `Invalid @assert/validate annotation - expected @assert/validate 2-tuple, but found tuple with length 1 (by schema.yml:3)`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(dataValuesYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+
+			dataValuesYAML2 := `#@data/values
+---
+#@assert/validate ("some validation", lambda v: v, "a message if valid", "a message if fail")
+foo: bar
+`
+			expectedErr = `Invalid @assert/validate annotation - expected @assert/validate 2-tuple, but found tuple with length 4 (by schema.yml:3)`
+
+			filesToProcess = files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(dataValuesYAML2))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+
+		})
+		t.Run("is a key=value pair", func(t *testing.T) {
+			opts := cmdtpl.NewOptions()
+			dataValuesYAML := `#@data/values
+---
+#@assert/validate foo=True
+foo: bar
+`
+
+			expectedErr := `Invalid @assert/validate annotation - expected @assert/validate to have 2-tuple as argument(s), but found keyword argument (by schema.yml:3)`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(dataValuesYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("has incorrect string args in a validation", func(t *testing.T) {
+			opts := cmdtpl.NewOptions()
+			dataValuesYAML := `#@data/values
+---
+#@assert/validate ({"foo":"bar"}, lambda v: v)
+foo: bar
+`
+
+			expectedErr := `Invalid @assert/validate annotation - expected @assert/validate to have string describing a valid value as the first item in the 2-tuple, but found type: dict (by schema.yml:3)`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(dataValuesYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("has incorrect function arg in a validation", func(t *testing.T) {
+			opts := cmdtpl.NewOptions()
+			dataValuesYAML := `#@data/values
+---
+#@assert/validate ("some validation", True)
+foo: bar
+`
+
+			expectedErr := `Invalid @assert/validate annotation - expected @assert/validate to have an assertion function as the second item in the 2-tuple, but found type: bool (by schema.yml:3)`
```

Comment:
Love these error messages. ❤️ 

In fact, many folks _are going to_ fumble their way through this syntax... these guards are going to be well-used, I predict. 💪🏻 

nit: suggesting a tweak.

```suggestion
			expectedErr := `Invalid @assert/validate annotation - expected second item in the 2-tuple to be an assertion function, but was a bool (at schema.yml:3)`
```